### PR TITLE
Added a StateStoreCleaner for cleaning up old state store files based on some retention

### PR DESCRIPTION
**Pull request by liyinan926**
Signed-off-by: Yinan Li <liyinan926@gmail.com>

### CODE REVIEW

**Comment by liyinan926** [comment]
@ydailinkedin can you help review this?

File: gobblin-metastore/src/main/java/gobblin/metastore/util/StateStoreCleaner.java [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+ * A utility class for cleaning up old state store files created by {@link gobblin.metastore.FsStateStore}
+ * based on a configured retention.
+ *
+ * @author ynli
+ */
+public class StateStoreCleaner {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(StateStoreCleaner.class);
+
+  private static final String STATE_STORE_CLEANER_RETENTION_KEY = "state.store.retention";
```
Comment (ydailinkedin) [inline gobblin-metastore/src/main/java/gobblin/metastore/util/StateStoreCleaner.java]: Can you change the key name to "state.store.retention.days"?

**Comment by liyinan926** [comment]
Updated cace7f4 to fix the dependencies.

**Comment by ydailinkedin** [comment]
LGTM.

**Pull request merged**